### PR TITLE
Wrap ZLow dialect into onnx_mlir::zlow namespace

### PR DESCRIPTION
**Pull request by etiotto**
This PR "wraps" the `ZLow` dialect and associated conversion passes into its namespace onnx_mlir:zlow.

Operationally this is accomplished by declaring the zhigh classes in a nested namespace as per best practices descrived in https://stackoverflow.com/questions/10816600/c-namespaces-how-to-use-in-header-and-source-files-correctly

Signed-off-by: Ettore Tiotto <etiotto@ca.ibm.com>

### CODE REVIEW

**Comment by tungld** [comment]
This is absolutely necessary. Given that the current NNPA code is not the latest code we have and not really works, it needs to be synced with our internal code, could we postpone this code change for a while until our movement is completed? Just try not to make the code in onnx-mlir too different from our internal code before we ensure everything functionally works well. Same to PR #1221

**Comment by etiotto** [comment]
@tungld yes it can be postponed. What is the ETA  to sync the NNPA code ?

**Comment by tungld** [comment]
@etiotto Thanks! Once PR #1219 is landed, we can do massive refactoring.

File: src/Accelerators/NNPA/Pass/DLCPasses.hpp
Comment (tungld) [inline src/Accelerators/NNPA/Pass/DLCPasses.hpp:23]: This is not used anymore. I removed it in PR #1219.

Comment (tungld) [reply src/Accelerators/NNPA/Pass/DLCPasses.hpp:23]: Could you please remove `createKrnlEmptyOptimizationPass` also? I think it is a dummy. Thanks!

Comment (etiotto) [reply src/Accelerators/NNPA/Pass/DLCPasses.hpp:23]: Will do thanks for pointing it out @tungld

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM, pending nit on namespace scope

File: src/Accelerators/NNPA/Conversion/ZLowToLLVM/ZLowToLLVMCommon.cpp
Comment (etiotto) [reply src/Accelerators/NNPA/Conversion/ZLowToLLVM/ZLowToLLVMCommon.cpp:28]: One alternative would be to "use" several using declarations like this: 
```
using namespace mlir;
namespace onnx_mlir;
namespace onnx_mlir::zlow;

void A::foo() { ... }
```

Now is unclear to which namespace the definition of `A::foo()` belongs to.  Wrapping the declaration of `A::foo()` in a namespace makes it clear that it belongs to the onnx_mlir::zlow namespace. 

See also: 
  - https://stackoverflow.com/questions/8681714/correct-way-to-define-c-namespace-methods-in-cpp-file
  - https://stackoverflow.com/questions/10816600/c-namespaces-how-to-use-in-header-and-source-files-correctly

Comment (etiotto) [reply src/Accelerators/NNPA/Conversion/ZLowToLLVM/ZLowToLLVMCommon.cpp:28]: Not sure if you mean we should nest  `onnx_mlir` inside `mlir`.   My opinion on that would be that we should not, because we are `using` MLIR but we are not part of MLIR (we aren't a project in tree).

Comment (AlexandreEichenberger) [reply src/Accelerators/NNPA/Conversion/ZLowToLLVM/ZLowToLLVMCommon.cpp:28]: I see places with 
```
using namespace mlir;
using namespace std;
```
that does not imply one included in the other ones.. it's just that the namespace is used for everything. I don't see the point of the `namespace {}` when it includes the whole file... if its for a subset, absolutely; if for the whole file, it seems to be a pedantic use of scope.

**Comment by etiotto** [comment]
@jenkins-droid test this please

**Pull request merged**
